cheatsheets_excluded/Content_Security_Policy_Cheat_Sheet.md

```diff
@@ -1,203 +1,199 @@
 
-`Content-Security-Policy: default-src 'none'; script-src 'self'; connect-src 'self'; img-src 'self'; style-src 'self';`
+In order to ensure backward compatibility, use the 2 directives in conjonction. Whenever a browser supports `report-to`, it will ignore `report-uri`. Otherwise, `report-uri` will be used.
 
-This policy allows images, scripts, AJAX, and CSS from the same origin, and does not allow any other resources to load (eg. object, frame, media, etc). (see <http://content-security-policy.com/>)
+## Special Directive Sources
 
-Mixed Content Policy
---------------------
+| Value           | Description                                                     |
+|-----------------|-----------------------------------------------------------------|
+| 'none'          | No URLs match.                                                  |
+| 'self'          | Refers to the origin site with the same scheme and port number. |
+| 'unsafe-inline' | Allows the usage of inline scripts or styles.                   |
+| 'unsafe-eval'   | Allows the usage of eval in scripts.                            |
 
-In order to prevent mixed content (resources being loaded over http, from a document loaded over https), one can use the value "https:" as a directive value.
+In case where the developer needs to use inline scripts, it's recommended to use `sha256` for the script or a `nonce` randomly generated on every page request. 
 
-For instance:
+For more details on hashes and nonces, check out [Scott Helme's Guide](https://scotthelme.co.uk/csp-cheat-sheet/#hashes).
 
-`Content-Security-Policy: default-src https:; connect-src https:; font-src https: data:; frame-src https:; `
-`img-src https: data:; media-src https:;  object-src https:; script-src 'unsafe-inline' 'unsafe-eval' https:; `
-`style-src 'unsafe-inline' https:;`
+# CSP Sample Policies
 
-This is what was used at Twitter, Oct 2014. The policy prevents mixed content, allows for scheme "data:" in font-src and img-src, allows for unsafe-inline and unsafe-eval for script-src, and unsafe-inline for style-src. (see: <https://twittercommunity.com/t/blocking-mixed-content-with-content-security-policy/26375>)
+## Basic CSP Policy
 
-Mixed Content has two categories: Active and Passive. Passive content consists of "resources which cannot directly interact with or modify other resources on a page: images, fonts, audio, and video for example", whereas active content is "content which can in some way directly manipulate the resource with which a user is interacting." (http://www.w3.org/TR/2014/WD-mixed-content-20140722)
+This policy will only allow resources from the originating domain for all the default level directives, and will not allow inline scripts/styles to execute. If your application and function with these restrictions, it drastically reduces your attack surface having this policy in place, and will work with most modern browsers.
```

Comment:
*If your application and function with these restrictions*: typo in sentence no?